src/api/l_data_sound.c

```diff
@@ -51,8 +51,9 @@ static int l_lovrSoundGetSampleRate(lua_State* L) {
 
 static int l_lovrSoundGetByteStride(lua_State* L) {
   Sound* sound = luax_checktype(L, 1, Sound);
-  uint32_t stride = lovrSoundGetStride(sound);
-  lua_pushinteger(L, stride);
+  size_t stride = lovrSoundGetStride(sound);
+  lovrCheck(stride < UINT32_MAX, "Sound contains impossibly many channels");
```

Comment:
I don't know, it feels really weird to me to check this on egress.  Like, I create a Sound and everything's great, but then I call a getter on it and it throws an error.  Also I'm not convinced that testing against `unit32_t` is correct here, since `lua_pushinteger` takes a `lua_Integer` which is a `ptrdiff_t`.

Reply:
1. This is, uh, a typo actually. I meant to add the check to a different function but I got a line number confused. Will remove.
2. The "check on egress" problem remains because of the below patch to sound:getFrames. I can fix that by switching to lua_optinteger instead of lua_optu32 if you're good with that.

    Overall there is a problem that we have API code that has inherent bounds or has unnecessary bounds due to the current typing, and we have data classes which are capable of representing anything up to SIZE_MAX, and it would make the most sense for either the data classes or the constructors to enforce a contract that they are no bigger than the "safe" size but there's no... like, there's no indication of who enforces those contracts or where. Maybe it could go in the header files? If the sizes can't be enforced by the type system I think it's best it at least be written down somewhere.
3. You're right about ptrdiff_t (again, this one change is outright a mistake) but there is a little bit of complexity there because any number which is returned to lua DOES have an upper bound and it's less than SIZE_MAX. PTRDIFF_MAX is usually lower than SIZE_MAX, and on luajit I would expect the upper bound to be something like 2^52 because it never(?) creates 32 bit integer types. I'm not sure if we need to worry about that (I can believe someone would load a 5 GB wav file either maliciously or in a pathological case. But 2^52 bytes is... 4 petabytes?) but if we choose to worry about it, I actually don't know how to query what the max value is.